app/models/assessment.rb

```diff
@@ -17,7 +17,8 @@ class Assessment < ApplicationRecord
   validates_presence_of :description
   validates_inclusion_of :is_hidden, in: [true, false]
   validates_presence_of :notes_count
-
+  validates :short_identifier, format: { with: /\A[a-zA-Z0-9\-_]+\z/,
+                                         message: 'short_identifier must be alphanumeric, hyphen, or underscore' }
```

Comment:
Right now this sentence isn't grammatically correct. Change to say "short_identifier must only contain..." and list character types.